### DISCUSSION TITLE
Filament runout sensor DIY

### DISCUSSION THREAD

**Discussion by Chrisyahya** [Jyers/Marlin#567] [Q&A] [answered]
Hello i noticed that jyers fw has filament runout sensor option. And i found a video (https://www.youtube.com/watch?v=5Jt-Qc67FDo)  about using roller switches DIY, can it work with jyers fw?
the thingiverse also available : https://www.thingiverse.com/thing:4563996

Since the switches only have 3 pins : NO, NC, C (normally open, normally closed, Common)
![image](https://user-images.githubusercontent.com/60070242/115828867-73a13280-a438-11eb-93a6-47ed7fb4a46a.png)

and the mainboard has 3 pins. I wonder where i should connect it..
![image](https://user-images.githubusercontent.com/60070242/115828773-59ffeb00-a438-11eb-8996-b79e2c6cd2b0.png)

i assume, C pins to G in mainboard, NC pins to VCC pins, NO pins to PA4?

**Comment by gothcha** [comment, marked as answer]
Hi, yes you can use this switch. I did exactly this. You should connect the common to PA4 and the NO to ground, you can leave the VCC unconnected (this is then safer if you happen to connect it wrongly). The firmware is configured for this polarity, but should it be required you can use the NC switch to ground to reverse the polarity. Good luck.

**Reply by Chrisyahya** [reply]
oh! thank you!

i thought i need the z limit PCB board.. so i can wire it directly without that pcb z limit board then.

gonna try it! Btw is the socket of that 3 pin named JST socket?

**Reply by Chrisyahya** [reply]
so here is this : 
connect COM to PA4
connect NO pins to Ground pins
leave the NC pins (not connected), also the vcc in board

if it doesnt work because some configuration firmware, 

disconnect NO pins, then connect NC pins to ground pins.

Am i correct?

**Reply by gothcha** [reply]
Yes this is all correct 👍 Good luck

**Comment by Jonibygg** [comment]
Hi, I do a little Hijack here. If I do a runout switch like this, is the precompiled .bin-files ready to use or do I need to change some settings and compile my self?

**Reply by Cal-LF** [reply]
@Jonibygg Jyers FW supports this type of filament runout setup. The BTT motion sensor is the type that requires setting changes.

**Reply by Jonibygg** [reply]
Thanks, that sounds easy even for me then. :-)
